pysignalscope/scope.py

```diff
@@ -853,7 +921,801 @@ def plot_channels(*channel: List['Scope'], timebase: str = 's', figure_size: Opt
+        HandleScope.chn_sel_button = Button(button_ax, 'Sel. Plot')
+        HandleScope.chn_sel_button.on_clicked(HandleScope.next_channel)  # Verknüpft die Schaltfläche mit der Funktion
+
+        # Button for reset slider position
+        button_ax = plt.axes([0.85, 0.1, 0.075, 0.05])  # Position and size of button
+        HandleScope.shsl_reset_button = Button(button_ax, 'Reset')
+        HandleScope.shsl_reset_button.on_clicked(HandleScope.reset_slider)  # Verknüpft die Schaltfläche mit der Funktion
+
+        # Textbox for step size
+        text_box_ax = plt.axes([0.6, 0.15, 0.2, 0.05])
+        HandleScope.shift_text_box = TextBox(text_box_ax, 'Shiftstep:', initial="1")
+        HandleScope.shift_text_box.on_submit(HandleScope.submit)
+
+        # Selected dataset (Show label)
+        labeltext = HandleScope.channelplotlist[0].get_label()
+        # Check, if label text is not set
+        if labeltext is None:
+            labeltext = "●"
+        else:
+            labeltext = "● "+labeltext
+        # Set labeltext in figure
+        HandleScope.selplotlabel = HandleScope.shiftfig.text(0.6, 0.3, labeltext, ha='left', va='top', fontsize=12)
+        HandleScope.selplotlabel.set_color(HandleScope.channelplotlist[HandleScope.chn_index].get_color())
+
+        # Register eventhandler
+        HandleScope.shiftfig.canvas.mpl_connect('button_press_event', HandleScope.on_press)
+        HandleScope.shiftfig.canvas.mpl_connect('motion_notify_event', HandleScope.on_motion)
+        HandleScope.shiftfig.canvas.mpl_connect('button_release_event', HandleScope.on_release)
+
+        # Slider for shift function
+        slider_ax = plt.axes([0.25, 0.1, 0.55, 0.03], facecolor="lightgoldenrodyellow")  # Position and size of sliders
+        HandleScope.shift_slider = Slider(slider_ax, 'Shift', -10.0, 10.0, valinit=0)  # Slider values
+        # Link slider with callback
+        HandleScope.shift_slider.on_changed(HandleScope.shiftchannel)
+        # Get the figure box object for check purpose
+        HandleScope.shiftfigbox = HandleScope.zoom_ax.get_window_extent()
+
+        # Log flow control
+        logging.debug(f"{class_modulename} :\nDisplay range x:{HandleScope.display_min_x},{HandleScope.display_max_x} " 
+                      f"Display range y:{HandleScope.display_min_y},{HandleScope.display_max_y}\n"                    
+                      f"Shift step x min,def,max:{HandleScope.min_shiftstep_x}, {def_shiftstep_x},{HandleScope.max_shiftstep_x}\n" 
+                      f"Shift step y min,def,max:{HandleScope.min_shiftstep_y}, {def_shiftstep_y},{HandleScope.max_shiftstep_y}\n" 
+                      f"Zoom area x,y:{HandleScope.zoom_delta_x}, {HandleScope.zoom_delta_y}")
+
+        plt.show()
+
+        # Return the list of channel shifts
+        return HandleScope.shiftlist
+
+    ##############################################################################
+    # Callback functions for interactive shifting of plots
```

Comment:
wording: functions -> methods
may search for other violations

Reply:
I agree, wording will be updated.
Done